bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/items/ManagedItemProvider.java

```diff
@@ -178,16 +130,18 @@ protected void addItemFactory(ItemFactory itemFactory) {
 
         if (failedToCreate.size() > 0) {
             // retry failed creation attempts
-            Iterator<ItemInfo> iterator = failedToCreate.iterator();
+            Iterator<String> iterator = failedToCreate.keySet().iterator();
             while (iterator.hasNext()) {
-                ItemInfo itemInfo = iterator.next();
-                Item item = itemFactory.createItem(itemInfo.itemType, itemInfo.itemName);
+                String itemName = iterator.next();
+                PersistedItem persistedItem = failedToCreate.get(itemName);
+                ActiveItem item = itemFactory.createItem(persistedItem.itemType, itemName);
                 if (item != null) {
-                    iterator.remove();
+                    failedToCreate.remove(itemName);
```

Comment:
Returns a Set view of the keys contained in this map. The set is backed by the map, so changes to the map are reflected in the set, and vice-versa. If the map is modified while an iteration over the set is in progress (except through the iterator's own remove operation), the results of the iteration are undefined.